corehq/apps/app_manager/static/app_manager/js/app_manager.js

```diff
@@ -279,83 +265,106 @@ hqDefine('app_manager/js/app_manager', function () {
-                    if ($(this).data('moduleuid') !== toModuleUid) {
-                        movingToNewModule = true;
-                        to = i;
-                        from = parseInt(ui.item.data('index'), 10);
-                        return false;
-                    }
-                });
+            var from, to;
+            if (movingFormToNewModule) {
+                [from, to] = calculateMoveFormToNewModule($sortable, ui, toModuleUid);
+            } else {
+                [from, to] = calculateMoveWithinScope($sortable);
             }
 
-            if (movingToNewModule || to !== from) {
-                var fromModuleUid = ui.item.data('moduleuid');
-                $form = $sortable.find('> .sort-action form');
-                $form.find('[name="from"], [name="to"]').remove();
-                $form.append('<input type="hidden" name="from" value="' + from.toString() + '" />');
-                $form.append('<input type="hidden" name="to"   value="' + to.toString() + '" />');
-                if (sortingForms) {
-                    $form.append('<input type="hidden" name="from_module_uid" value="' + fromModuleUid + '" />');
-                    $form.append('<input type="hidden" name="to_module_uid"   value="' + toModuleUid + '" />');
-                }
-
+            if (to !== from || movingFormToNewModule) {
+                var $form = $sortable.find('> .sort-action form');
+                saveDataToForm($form, from, to, fromModuleUid, toModuleUid, sortingForms);
                 resetIndexes($sortable);
-                if (fromModuleUid !== toModuleUid) {
-                    var $parentSortable = $sortable.parents(".sortable"),
-                        $fromSortable = $parentSortable.find("[data-uid=" + fromModuleUid + "] .sortable");
-                    resetIndexes($fromSortable);
+                if (movingFormToNewModule) {
+                    resetOldModuleIndices($sortable, fromModuleUid);
                 }
-                $.ajax($form.attr('action'), {
-                    method: 'POST',
-                    data: $form.serialize(),
-                    success: function () {
-                        hqImport('hqwebapp/js/alert_user').alert_user(gettext("Moved successfully."), "success");
-                    },
-                    error: function (xhr) {
-                        hqImport('hqwebapp/js/alert_user').alert_user(xhr.responseJSON.error, "danger");
-                    },
-                });
+                submitReorderForm($form);
                 hqImport("app_manager/js/menu").setPublishStatus(true);
             }
         }
-
+        function calculateMoveFormToNewModule($sortable, ui, toModuleUid) {
+            var from = -1, to = -1;
+            $sortable.children().not('.sort-disabled').each(function (i) {
+                if ($(this).data('moduleuid') !== toModuleUid) {
+                    to = i;
+                    from = parseInt(ui.item.data('index'), 10);
```

Comment:
Is the `parseInt` necessary? Usually jquery's `data` will parse numbers out automatically.

Reply:
Not sure, these were here already